conclave-host/src/main/kotlin/com/r3/conclave/host/internal/SocketHostEnclaveInterface.kt

```diff
@@ -24,13 +23,10 @@ import java.util.concurrent.Semaphore
 class SocketHostEnclaveInterface(port: Int = 0, private val maxConcurrentCalls: Int) : HostEnclaveInterface(), Closeable {
     private val serverSocket = ServerSocket(port)
```

Comment:
The socket shouldn't be bound in the c'tor. In fact this socker is used and then closed in `start` you can move this there, and then have the `port` property as a lateinit which is set in `start`.

Reply:
We also need to pass the port to the enclave process, and we don't know what the port will be until it's bound, so there's that too.

Reply:
Yes `start` should probably be split into two, `start` and `run`.

Reply:
Sorry, I thought `start` blocked, but it doesn't. So no, I'm not sure why we would need a separate method. Looking at GramineEnclaveHandle, you would simply call `start` synchronously. I'm not sure why it needs to be done on a separate thread.

Reply:
>We also need to pass the port to the enclave process, and we don't know what the port will be until it's bound, so there's that too.

I think this can be fixed if we say that the port is only defined after `start` is called. Then the can use pass it to the gramine process. This would require not starting the interface on a separate thread, which I don't think seems to be necessary.

Reply:
> Looking at GramineEnclaveHandle, you would simply call `start` synchronously. I'm not sure why it needs to be done on a separate thread.

Start blocks until the enclave connects, so it has to be called either asynchronously or after the enclave process has been started. In the current implementation (binding the port at construction), you're right, it probably doesn't need to be done in a thread like this!

Reply:
I've cleaned this up a bit, adding a separate `bindPort()` method and removing the unneeded thread in the enclave handle.
LMK if you think this is sufficient.